src/components/Amalgamation/AmalgamationStatement.vue

```diff
@@ -0,0 +1,119 @@
+<template>
+  <v-card
+    id="amalgamation-statement"
+    flat
+  >
+    <v-row>
+      <v-col
+        cols="12"
+        sm="3"
+        class="pr-4"
+      >
+        <label>
+          Amalgamation Statement
+        </label>
+      </v-col>
+
+      <v-col
+        cols="12"
+        sm="9"
+      >
+        <v-radio-group
+          v-model="courtApproval"
+          class="mt-0 pt-0"
+          @change="setCourtApproval()"
+        >
+          <v-radio
+            class="radio-button"
+            label="With Court Approval"
+            :value="true"
+          />
+          <div
+            class="ml-8 statement-text"
+          >
+            This amalgamation has been approved by the court and a copy of the
+            entered court order approving the amalgamation has been obtained
+            and has been deposited in the records office of each of the
+            amalgamating companies.
+          </div>
+
+          <v-radio
+            class="pt-4 radio-button"
+            label="Without Court Approval"
+            :value="false"
+          />
+          <div
+            class="ml-8 statement-text"
+          >
+            This amalgamation has been effected without court approval.
+            A copy of all of the required affidavits under section 277(1)
+            have been obtained and the affidavit obtained from each
+            amalgamating company has been deposited in that company’s
+            records office.
+          </div>
+        </v-radio-group>
+      </v-col>
+    </v-row>
+  </v-card>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Component, Emit } from 'vue-property-decorator'
+import { AmalgamationStateIF } from '@/interfaces/store-interfaces/state-interfaces/amalgamation-state-interface'
+import { Getter } from 'pinia-class'
+import { useStore } from '@/store/store'
+
+@Component({})
+export default class AmalgamationStatement extends Vue {
+  @Getter(useStore) getAmalgamationCourtApproval!: boolean
+
+  // Local properties
+  courtApproval: AmalgamationStateIF['courtApproval'] = null
```

Comment:
This is a clever way to specify the type but I'm not sure if I love it. Did you see somewhere that it is a best practise?

The issue I have with this is that you need to know the property name in the interface, but I don't know if that's worse than just saying it's a boolean (which you need to know in this file anyway).

Another thing I'm not sure I like is the syntax, which looks like a quoted object property (compared to a named object property). There are lint rules (strict-property-naming) to enforce the named style, and lots to read on the subject... My conclusion is that quoted properties are to be avoided,

(This is not the same as having object properties in quotes. That's another topic.)

Reply:
> Did you see somewhere that it is a best practise?

Nope, I just go through other files and find that it makes me more clear to write down the property name at here.

> but I don't know if that's worse than just saying it's a boolean

I am thinking should I simply write `courtApproval = null as boolean` or do `courtApproval: AmalgamationStateIF['courtApproval'] = null as boolean`? The first one is more simple, the second one may contain more information but don't know how to avoid the quoted properties

Reply:
Yes, exactly.

Let's leave your current code for now, use it a bit, and decide later. It's easy to change later if we decide against it.

Reply:
Shaoyun, you can resolve this conversation after you've read this.